src/sqlfluff/core/parser/grammar/base.py

```diff
@@ -413,7 +417,7 @@ def _trim_elem(seg):
 
         # Make some buffers
         seg_buff = segments  # pragma: no cover
```

Comment:
Could the lines from here to the end of the function be removed? (They are all marked `no cover`.)

Reply:
I wondered that myself but I think was too chicken to try it. If we're both thinking it I'll try!

Reply:
Hmmmm - actually I think it's more complicated than that - I'll make a separate issue for it though, to talk through things. Great idea though.

Reply:
new issue is live: #3721 Let's handle that there.